docs: SDK architecture + small changes (#435)
GetStream/stream-video-js: packages/react-sdk/docusaurus/docs/React/01-basics/03-sdk-architecture.mdx

title: SDK Architecture
---

This page describes the main building blocks of the SDK.

## Call Engine

> The call engine is made up of two parts:

Reviewer: ```suggestion
We understand the term **call engine** as a set of the following functionalities:
```

- The client lets you interact with Stream's video API
- Various utility hooks and contexts that let you access the state easily and give high-level APIs for common video/audio call-related tasks such as device management

> The SDK API supports two different call lifecycles. We have prepared a  [Call lifecycle guide](../02-guides/02-call-lifecycle.gen.mdx) to help you identify the right scenario for your project.

Reviewer: ```suggestion
The SDK API supports two different call lifecycles. We have prepared a [Call lifecycle guide](../02-guides/02-call-lifecycle.gen.mdx) to help you identify the right scenario for your project.
```

> Additionally, there are various available call types with configurations suiting different use cases (for example, livestream or audio room). If you want to know more about these, visit our [Configuring call defaults](./06-configuring-call-defaults.mdx) guide.

Reviewer: What does "call types with configurations" mean?

Reviewer: ```suggestion
Additionally, there are various available call types configurations suiting different use cases (for example, livestream or audio room). If you want to know more about these, visit our [Configuring call defaults](./06-configuring-call-defaults.mdx) guide.
```
